chrome-extension/popup.js

```diff
@@ -0,0 +1,90 @@
+const API_KEY = 'AIzaSyAed5yuxCNrXlEFAqSoWwAQpE3Ng95Tzl8';
+const DISCOVERY_DOCS = [
+  "https://docs.googleapis.com/$discovery/rest?version=v1",
+  "https://www.googleapis.com/discovery/v1/apis/drive/v3/rest"
+];
+
+const NOTES_TEMPLATE_DOC_ID = '1XlcAy-vrleXBxJl5Qy_SxGUyTqcwdUIhyJI2BygpNEc';
+const REVOKE_TOKEN_URL = 'https://accounts.google.com/o/oauth2/revoke?token=';
+
+let loggedIn = false;
+
+/**
+ * Initializes Google APIs client library for browser-side JavaScript (gapi)
+ * and the button functions when script is loaded.
+ */
+function onGAPILoad() {
+  document.getElementById('generate-note-btn').onclick = generateNote;
+  document.getElementById('logout-btn').onclick = logout;
+  document.getElementById('doc-name-input').value = 'gNote ' + getDate();
+
+  gapi.client.init({
+    apiKey: API_KEY,
+    discoveryDocs: DISCOVERY_DOCS,
+  }).then(function() {
+    chrome.identity.getAuthToken({interactive: true}, function(token) {
+      if (chrome.runtime.lastError) {
+        return;
+      }
+      gapi.auth.setToken({
+        access_token: token,
+      });
+      loggedIn = true;
+      setAccountInfo();
+    })
+  }, function(error) {
+    console.log('error', error)
+  });
+}
```

Comment:
This looks good but I think could be more readable if some functionality is extracted out to a helper function and if we do error handling for the promise in a catch block. I haven't tested what I've suggested below but I think it should work.

```suggestion
  gapi.client.init({
    apiKey: API_KEY,
    discoveryDocs: DISCOVERY_DOCS,
  }).then(handleLogin)
  .catch(error => console.log('Error:', error));
}
  
function handleLogin() {
  chrome.identity.getAuthToken({interactive: true}, function(token) {
      if (chrome.runtime.lastError) {
        return;
      }
      gapi.auth.setToken({
        access_token: token,
      });
      loggedIn = true;
      setAccountInfo();
    });
}
```

Reply:
You're right, that's definitely more readable!